docs/VSCODE.md

```diff
@@ -0,0 +1,93 @@
+# Developing with VisualStudio Code
+
+We recommend our contributors to use [VisualStudio Code](https://code.visualstudio.com/) with the following extensions installed:
+ - [Prettier - Code Formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) for automatic formatting of source files on save.
+ - [TSLint](https://marketplace.visualstudio.com/items?itemName=eg2.tslint) to highlight and auto-fix linting problems directly in the editor.
+
+Our monorepo comes with few preconfigured [VSCode
+Tasks](https://code.visualstudio.com/docs/editor/tasks):
+
+ - The build task is configured to run the TypeScript compiler
+ - The test task is configured to run `npm test` (which runs the build before running any tests).
+
+## How to verify TypeScript setup
+
+### Compilation errors
+
+1. Open any existing TypeScript file, e.g. `packages/src/index.ts`
+
+2. Add a small bit of code to break TypeScript's type checks, for example:
+
+    ```ts
+    const foo: number = 'bar';
+    ```
+
+3. Verify that VS Code editor has marked `foo` with a red underscore. Hover over `foo` and check the problem message. It should mention `[ts]` source, e.g.
+
+    ```text
+    [ts] Type '"bar"' is not assignable to type 'number'.
+
+    ```
+
+4. Check VS Code's [PROBLEMS Window](https://code.visualstudio.com/docs/getstarted/tips-and-tricks#_errors-and-warnings).  There should be an entry showing the same tslint error. When you click on the entry, it should jump on the problematic line.
+
+5. Close the editor tab. (This will clear the PROBLEMS entry reported by TSLint extension).
+
+6. Run the test task ("Tasks: Run test task"). This will invoke package scripts like `npm test` under the hood.
+
+7. Open "Tasks" OUTPUT window and verify that compilation error was parsed by VSCode.
+
+8. Verify that compilation errors are correctly associated with the problematic source code line.
+   _(2018-02-82: this is does not work now, `tsc` is reporting paths relative to individual package directories.)_
```

Comment:
typo if `2018-02-82` is a date here. I suggest we link it to an issue, but if it's fixed, we can get rid of this message.

Reply:
Will fix. I opened https://github.com/strongloop/loopback-next/issues/1010 to keep track of this bug.